datadog_checks_dev/tests/tooling/configuration/test_load.py

```diff
@@ -0,0 +1,2776 @@
+# (C) Datadog, Inc. 2019
```

Comment:
(nits) 

1. This file is quite long. Maybe we can split into multiple smaller files.

2. Some cases are quite suitable for pytest parametrized tests. That will make it easier to add new tests, ease maintenance.